pulley/src/interp.rs

```diff
@@ -101,14 +114,34 @@ impl Vm {
                 },
             }
         }
+    }
 
-        self.run(func)?;
+    /// Peforms the internal part of [`Vm::call`] where bytecode is actually
+    /// executed.
+    pub unsafe fn call_run(&mut self, pc: NonNull<u8>) -> DoneReason<()> {
+        self.state.debug_assert_done_reason_none();
+        let interpreter = Interpreter {
+            state: &mut self.state,
+            pc: UnsafeBytecodeStream::new(pc),
+        };
+        let done = interpreter.run();
+        self.state.done_decode(done)
+    }
+
+    /// Peforms the tail end of [`Vm::call`] by returning the values as
+    /// determined by `rets` according to Pulley's ABI.
+    pub unsafe fn call_end<'a>(
```

Comment:
```suggestion
    /// Peforms the tail end of [`Vm::call`] by returning the values as
    /// determined by `rets` according to Pulley's ABI.
    ///
    /// # Unsafety
    ///
    /// In addition to the invariants documented for `call`, this may
    /// only be called after `call_run`.
    pub unsafe fn call_end<'a>(
```